src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/args/Args.java

```diff
@@ -108,16 +126,32 @@
   }
 
   /**
-   * @return [--name, file.getAbsolutePath()] or [] if file=null.
+   * Produces the flag form of a file object, using {@link File#toPath()}.
+   *
+   * @return {@code [--name, file.toPath().toString()]} or {@code []} if file is null.
    */
   static List<String> filePath(String name, File file) {
-    if (file != null && !Strings.isNullOrEmpty(file.getAbsolutePath())) {
-      return Arrays.asList("--" + name, file.getAbsolutePath());
+    if (file != null) {
+      return path(name, file.toPath());
     }
     return Collections.emptyList();
   }
 
   /**
+   * Produces the flag form of a path object, using {@link Path#toString()}.
+   *
+   * @return {@code [--name, path.toString()]} or {@code []} if path is null or not set.
+   */
+  static List<String> path(String name, Path path) {
```

Comment:
That would defeat the purpose of retrieving the File/Path contents the right way..
Also, this way, an interface is provided for those who want to start building features in the library that adhere to Java file system operations best practices.